recipes/libgcrypt/all/conanfile.py

```diff
@@ -0,0 +1,77 @@
+import os
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.33.0"
+
+class LibgcryptConan(ConanFile):
+    name = "libgcrypt"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.gnupg.org/download/index.html#libgcrypt"
+    description = "Libgcrypt is a general purpose cryptographic library originally based on code from GnuPG"
+    topics = ("conan", "libgcrypt", "gcrypt", "gnupg", "gpg", "crypto", "cryptography")
+    license = "LGPL-2.1-or-later"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+    requires = 'libgpg-error/1.36'
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def configure(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("This recipe only support Linux. You can contribute Windows and/or Macos support.")
```

Comment:
Move this check to `validate`.

Reply:
Why didn't you demand this to be moved in a different method in libgpg-error PR? Don't we tend to conform some general idea in conanfile.py appearance?

Reply:
Because some PR a so huge and/or complicated we miss things ... even when four of us put eyes.

https://github.com/conan-io/conan-center-index/pull/6863

Reply:
@prince-chrismc, done